ci_framework/roles/ci_setup/tasks/packages.yml

```diff
@@ -48,3 +48,9 @@
         remote_src: true
         mode: "0755"
         creates: "{{ cifmw_ci_setup_openshift_client_install_path }}/oc"
+
+- name: Inject oc completion in local profile
+  ansible.builtin.shell:  # noqa: risky-shell-pipe
+    cmd: >-
+      {{ cifmw_ci_setup_openshift_client_install_path }} completion bash |
```

Comment:
This one looks solid, however there is one concern.
What's gonna happen if an user is going to run `reproducer` on the localhost (or remote host for that matter) multiple times? Is it a part of "non-ci workflow"?

Reply:
This is being executed [here](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/playbooks/01-bootstrap.yml#L40-L44)
The reproducer uses `bootstrap` and _I think_ it might be used many times when trying to reproduce  jobs [description here](https://github.com/openstack-k8s-operators/ci-framework/blob/main/docs/source/reproducers/03-zuul.md?plain=1#L180-L183)

Reply:
we may generate the completion in another file, and ensure with a simple lineinfile in .bash_profile that we do source it. shouldn't be too hard.

That said - this one will run twice indeed - once on the hypervisor (because we consume that role) and once on the controller-0. The hypervisor may therefore get a growing profile.
Of course, best thing would be to get `oc` deployed in global location, allowing to get a global completion in /etc/profile.d :).